### PR TITLE
Revisit storing using bytes directly in IBC packets

### LINKED ISSUE #1517: Revisit storing using bytes directly in IBC packets
Try using it instead of hex, if it does not work. Find out why and maybe need to submit proper PRs to Cosmos-SDK.

### PR DESCRIPTION
**Pull request by Benzbeeb**
Fixed #1517

### CODE REVIEW

File: bridges/evm/test/Bridge.test.js [hunk 1]
```diff
@@ -195,53 +195,60 @@ contract("Bridge", ([_, owner, alice, bob]) => {
...
 
     it("should not accept unrelayed block", async () => {
       await expectRevert(
         this.bridge.verifyOracleData(
-          "9999", // _blockHeight
-          ["band test", 1, "030000004254436400000000000000", 4, 4], // _requestPacket
-          ["band test", 1, 4, 1587734008, 1587734012, 1, "d8720b0000000000"], // _responsePacket
-          "326", // _version
+          "999999", // _blockHeight
+          ["beeb", 1, "0x030000004254436400000000000000", 4, 4], // _requestPacket
```
Comment (taobun) [inline bridges/evm/test/Bridge.test.js:207]: We need to change client id look more professional? @sorawit

File: chain/client/proof/packets_test.go [hunk 2]
```diff
@@ -9,25 +10,34 @@ import (
 )
 
 func TestCalculateResultHash(t *testing.T) {
-	// RawByte is d9c589270a0962616e64207465737410011a1e30333030303030303432353434333634303030303030303030303030303020042804
+	// RawByte is d9c589270a046265656210011a0f03000000425443640000000000000020012801
+	var err error
+	calldata, err := hex.DecodeString("030000004254436400000000000000")
```
Comment (taobun) [inline chain/client/proof/packets_test.go]: We have `mustDecodeString` in util

**Review by taobun — CHANGES_REQUESTED** [review]

File: bridges/evm/test/Bridge.test.js (hunk 1 above)
Comment (sorawit) [reply bridges/evm/test/Bridge.test.js:207]: I think this is ok. nothing rude or unappropriate.

**Review by taobun — DISMISSED** [review]

File: chain/x/oracle/types/types.proto [hunk 3]
```diff
@@ -200,7 +200,7 @@ message OracleRequestPacketData {
   ];
   // Calldata is the hex-encoded of the calldata bytes available for oracle
```
Comment (sorawit) [inline chain/x/oracle/types/types.proto]: Please update the comment.

File: chain/x/oracle/keeper/request.go [hunk 4]
```diff
@@ -113,15 +113,15 @@ func (k Keeper) ResolveRequest(
 		types.EventTypeRequestExecute,
 		sdk.NewAttribute(types.AttributeKeyClientID, req.ClientID),
 		sdk.NewAttribute(types.AttributeKeyOracleScriptID, fmt.Sprintf("%d", req.OracleScriptID)),
-		sdk.NewAttribute(types.AttributeKeyCalldata, req.Calldata),
+		sdk.NewAttribute(types.AttributeKeyCalldata, hex.EncodeToString(req.Calldata)),
```
Comment (sorawit) [inline chain/x/oracle/keeper/request.go]: I think it's more conventional to just wrap `string`

**Review by sorawit — CHANGES_REQUESTED** [review]

File: chain/x/oracle/types/types.proto [hunk 5]
```diff
@@ -238,9 +238,9 @@ message OracleResponsePacketData {
   // ResolveStatus is the status of this oracle request, which can be OK, ERROR,
   // or EXPIRED.
   ResolveStatus resolve_status = 6;
-  // Result is the hex-encoded of the final aggregated value only available if
+  // Result is the final aggregated value only available if
```
Comment (taobun) [inline chain/x/oracle/types/types.proto]: It can combine to one line?

**Review by taobun — CHANGES_REQUESTED** [review]

**Review by taobun — APPROVED** [review]

**Review by sorawit — APPROVED** [review]

**Pull request merged**